src/expertsystem/amplitude/helicity_decay.py

```diff
@@ -444,14 +442,11 @@ def __generate_intensities(
                 graph_group
             )
             incoherent_intensity.intensities.append(coherent_intensity)
-        if len(incoherent_intensity.intensities) > 1:
-            intensity: IntensityNode = NormalizedIntensity(
-                incoherent_intensity
-            )
-        else:
+        if len(incoherent_intensity.intensities) == 1:
```

Comment:
I'm thinking what happens in the case when the `len` is 0. In that sense, the following might be better:
```python
if len(incoherent_intensity.intensities) == 0:
    raise ValueError("List of incoherent intensities cannot be empty")
if len(incoherent_intensity.intensities) == 1:
    return incoherent_intensity.intensities[0]
return incoherent_intensity
```
(where the first `if` statement could be removed if that is a redundant check)

Reply:
Good point

Reply:
d4aaaec